service/controller/clusterapi/v27/detection/detection.go

```diff
@@ -46,11 +46,11 @@ func (d *Detection) ShouldScale(ctx context.Context, cr v1alpha1.AWSConfig) (boo
 	}
 
 	if !cc.Status.TenantCluster.TCCP.ASG.IsEmpty() && cc.Status.TenantCluster.TCCP.ASG.MaxSize != legacykey.ScalingMax(cr) {
-		d.logger.LogCtx(ctx, "level", "debug", "message", "detected the tenant cluster should scale due to scaling max changes")
+		d.logger.LogCtx(ctx, "level", "debug", "message", "detected the tenant cluster should scale due to scaling max changes: cc.Status.TenantCluster.TCCP.ASG.MaxSize is %d while legacykey.ScalingMax(cr) is %d", cc.Status.TenantCluster.TCCP.ASG.MaxSize, legacykey.ScalingMax(cr))
```

Comment:
I would skip `cc` in those messages like:
```suggestion
		d.logger.LogCtx(ctx, "level", "debug", "message", "detected the tenant cluster should scale due to scaling max changes: .Status.TenantCluster.TCCP.ASG.MaxSize is %d while legacykey.ScalingMax(cr) is %d", cc.Status.TenantCluster.TCCP.ASG.MaxSize, legacykey.ScalingMax(cr))
```

Also using function names from key is cryptic but I think it's ok first step.

I also started discussion to address that: https://gigantic.slack.com/archives/C2MS2VB37/p1556637615023500

Reply:
Why would you remove `cc` when that's reference to our internal context? That `.Status...` prefix makes me intuitively think about CR object which this is not.

Reply:
> makes me intuitively think about CR object

That was my intention. I think those paths are the same. I'd prefer to see decision made based on the CR. That is more understandable for someone not up to date with the codebase.

Reply:
> I think those paths are the same. I'd prefer to see decision made based on the CR.

Those are quite different from CR. The paths in CR are:
* `.Status.Cluster.Scaling.DesiredCapacity`
* `.Spec.Cluster.Scaling.Max`

Also: `cc` stands for `controllercontext` which is the internal variable that's used to hold state from cloud provider. I don't understand how the message even ideally could be about having difference between CR vs. CR?

Maybe we should hang on this as I think we are talking about different things possibly?

Reply:
Oh yeah. Not everything in cc comes from the CR status. Fine then. Let's go!